server/src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -1088,86 +964,16 @@ protected E lockAndLoadById(Class<E> entityClass, Serializable id) {
-            PersistenceContext context = session.getPersistenceContext();
-
-            SortedSet<Serializable> idSet = new TreeSet<>();
-            SortedMap<Serializable, E> entitySet = new TreeMap<>();
+        SortedSet<Serializable> idSet = new TreeSet<>();
 
-            // Step through the collection of IDs and figure out which entities we have to refresh,
-            // and which we need to lookup.
-            for (Serializable id : ids) {
-                // Make sure we don't doubly load/lock anything
-                if (id != null && !idSet.contains(id) && !entitySet.containsKey(id)) {
-                    EntityKey key = session.generateEntityKey(id, persister);
-                    E entity = (E) context.getEntity(key);
-
-                    if (entity != null) {
-                        EntityEntry entry = context.getEntry(entity);
-
-                        // Make sure the entry hasn't been deleted or otherwise removed.
-                        if (entry.isExistsInDatabase() && entry.getStatus() != Status.DELETED &&
-                            entry.getStatus() != Status.GONE) {
-
-                            entitySet.put(id, entity);
-                        }
-                    }
-                    else {
-                        idSet.add(id);
-                    }
-                }
-            }
-
-            // First address the slow (and hopefully smaller) part of the lookup and refresh the
-            // entities that exist
-
-            // TODO: Maybe add a debug warning here to call out situations where our existing
-            // entity size is larger than our absent entity size. Those are areas where we may be
-            // reloading entities unnecessarily and could be optimized to avoid doing extraneous
-            // work.
-            if (entitySet.size() > 0) {
-                for (E entity : entitySet.values()) {
-                    entityManager.refresh(entity, LockModeType.PESSIMISTIC_WRITE);
-                    result.add(entity);
-                }
-            }
-
-            // Fetch the remaining entities from the DB
-            if (idSet.size() > 0) {
-                result.addAll(((Session) session)
-                    .byMultipleIds(this.entityType())
-                    .with(new LockOptions(LockMode.PESSIMISTIC_WRITE))
-                    .with(CacheMode.REFRESH)
-                    .enableSessionCheck(false)
-                    .multiLoad(new ArrayList(idSet)));
-            }
+        for (Serializable id : ids) {
+            idSet.add(id);
         }
 
-        // Should we be returning a view of the list, rather than the fully mutable list here?
-        return result;
+        return this.currentSession()
+            .byMultipleIds(this.entityType())
+            .with(new LockOptions(LockMode.PESSIMISTIC_WRITE))
+            .multiLoad(new ArrayList(idSet));
```

Comment:
Also, is there a reason we don't want the `.with(CacheMode.REFRESH)` as in the `lockAndLoadById` method?